SpatialGDK/ExternalSchemaCodegen/Programs/Improbable.CodeGen.Unreal/EnumGenerator.cs

```diff
@@ -0,0 +1,50 @@
+using Improbable.CodeGen.Base;
+using System;
+using System.Linq;
+using System.Text;
+
+namespace Improbable.CodeGen.Unreal
+{
+    static class EnumGenerator
+    {
+        private static string DefaultEnumHeaderIncludes = string.Join(Environment.NewLine, new[]{
+            "#include <cstdint>",
+            "#include <functional>",
+        });
+
+        public static string GenerateTopLevelEnum(EnumDefinition enumDefinition, Bundle bundle)
+        {
+            var enumNamespace = Text.GetNamespaceFromTypeName(enumDefinition.Identifier.QualifiedName);
+            var builder = new StringBuilder();
+
+            builder.AppendLine($"// Generated by {UnrealGenerator.GeneratorTitle}");
+            builder.Append(Environment.NewLine);
+
+            builder.AppendLine($"#pragma once");
+            builder.Append(Environment.NewLine);
+
+            builder.AppendLine(DefaultEnumHeaderIncludes);
+            builder.Append(Environment.NewLine);
+
+            builder.AppendLine(string.Join(Environment.NewLine, enumNamespace.Select(t => $"namespace {t} {{")));
```

Comment:
Is the namespace enum the convention we follow? I see `enum class` being used more often in the engine for new enums.

Reply:
not sure what you mean here?

Reply:
[Why is enum class preferred over plain enum? (stackoverflow)](https://stackoverflow.com/questions/18335861/why-is-enum-class-preferred-over-plain-enum)

I think wrapping enums in a namespace used to be the solution to the problem with plain enums described in the linked post. However, in more recent standard this has been replaced by enum class and I see this being used in the engine, too. They decided not to refactor all of their existing enums so that things don't break.

Reply:
I'm just raising a question about what should the convention be in the code we generate for enums.

Reply:
Discussed offline - the generated code already uses `enum class` which is good, I misunderstood what will be generated at first.